### PR TITLE
Apply `sample_for_compression` to all block-based tables

### PR DESCRIPTION
**Pull request by ajkr**
Previously it only applied to block-based tables generated by flush. This restriction
was undocumented and blocked a new use case. Now compression sampling
applies to all block-based tables we generate when it is enabled.

Test Plan: new unit test

### CODE REVIEW

File: db/db_properties_test.cc [hunk 1]
```diff
@@ -1175,6 +1175,60 @@ class CountingDeleteTabPropCollectorFactory
...
+    return Status::OK();
+  }
+
+  Status AddUserKey(const Slice& /*user_key*/, const Slice& /*value*/,
+                    EntryType /*type*/, SequenceNumber /*seq*/,
+                    uint64_t /*file_size*/) override {
+    return Status::OK();
+  }
+
+  void BlockAdd(uint64_t /* blockRawBytes */, uint64_t blockCompressedBytesFast,
```
Comment (riversand963) [inline db/db_properties_test.cc:1198]: Nit: block_compressed_bytes_fast and block_compressed_bytes_slow?
To make the linter happy, we need to update the variable name in `TablePropertiesCollector::BlockAdd()` too.

**Review by riversand963 — APPROVED** [review]
LGTM.

File: db/db_properties_test.cc (hunk 1 above)
Comment (ajkr) [reply db/db_properties_test.cc]: Done.

**Review by ajkr — COMMENTED** [review]
Thanks for the review!

**Pull request closed without merging**